common/src/serial.rs

```diff
@@ -17,7 +17,12 @@ impl SerialPort {
 
 impl fmt::Write for SerialPort {
     fn write_str(&mut self, s: &str) -> fmt::Result {
-        self.port.write_str(s).unwrap();
+        for char in s.bytes() {
+            match char {
+                b'\n' => self.port.write_str("\r\n").unwrap(),
+                byte => self.port.send(byte),
+            }
+        }
```

Comment:
Another fix for #473 might be to change the `writeln!` call to a `write("...\r\n")` call, this wouldn't work if the log message contains multi-line strings, so I think this solution is preferable. 

Performance could certainly be improved a bit, but this code is not at all performance-critical, so let's keep things simple.